src/wallet/test/rpc_wallet_tests.cpp

```diff
@@ -667,12 +733,14 @@ BOOST_AUTO_TEST_CASE(rpc_wallet_z_importexport)
         myaddrs.insert(keyIO.EncodePaymentAddress(pwalletMain->GenerateNewSproutZKey()));
     }
 
-    // Verify number of addresses stored in wallet is n1+n2
+    // Verify number of addresses stored in wallet is correct
     int numAddrs = myaddrs.size();
-    BOOST_CHECK(numAddrs == (2 * n1) + n2);
-    pwalletMain->GetSproutPaymentAddresses(addrs);
+    // sprout_addrs + sapling addrs with spend authority + new sprout addrs
+    BOOST_CHECK(numAddrs == (n1 + (n1 / 2) + n2));
+    pwalletMain->GetSproutPaymentAddresses(sproutAddrs);
     pwalletMain->GetSaplingPaymentAddresses(saplingAddrs);
-    BOOST_CHECK(addrs.size() + saplingAddrs.size() == numAddrs);
+    // here we also include the watchonly sapling addresses
+    BOOST_CHECK(sproutAddrs.size() + saplingAddrs.size() == numAddrs + (n1 / 2));
 
     // Ask wallet to list addresses
     BOOST_CHECK_NO_THROW(retValue = CallRPC("z_listaddresses"));
```

Comment:
In other cases where we have `foo` and `z_foo` methods, `foo` is the legacy one (e.g. `sendmany` and `z_sendmany`). Should we add a note to the `z_listaddresses` method doc pointing to `listaddresses` as a method that can provide more information?

Reply:
Well, we'd have needed to merge it with `z_listaddresses`, or use a completely different name. Sorry for not spotting this earlier.